### PR TITLE
Refactor Output

### PR DESCRIPTION
**Pull request by bencollins54**
- Change output flag to accept csv, json & table
- Add Comfy Table to handle cli table output.
- Minor filewalk tweaks
- Fix linked project on Linux.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "rust-analyzer.linkedProjects": [".\\crates\\fta-wasm\\Cargo.toml"]
-}
+  "rust-analyzer.linkedProjects": ["./crates/fta-wasm/Cargo.toml"],
+  "[rust]": {
+      "editor.defaultFormatter": "rust-lang.rust-analyzer", // Makes the magic
```
Comment (sgb-io) [inline .vscode/settings.json:4]: comments in json - naughty

File: crates/fta/src/main.rs [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
 struct Cli {
     project: String,
 
     // Output JSON output
-    #[arg(long)]
-    json: bool,
+    #[arg(long, short, default_value = "table")]
+    format: String,
```
[the remark this answers is not in the document]
Comment (sgb-io) [reply crates/fta/src/main.rs:13]: - Keep the `json` bool
- Also add the new `format` string. Values: `json`, `csv`, `table`
- `json` `true` is the same result as `format` `'json'`
- If `format` is supplied, and `!= json`, but `json` is also `true`, panic (invalid combination of options)

File: crates/fta/src/output/tests.rs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#[cfg(test)]
```
Comment (sgb-io) [inline crates/fta/src/output/tests.rs:1]: tests missing - either add some, or remove this file

File: crates/fta/Cargo.toml [hunk 4]
```diff
@@ -23,4 +24,4 @@ swc_common = "0.31.12"
 swc_ecma_ast = "0.104.5"
 swc_ecma_parser = "0.134.12"
 swc_ecma_visit = "0.90.5"
-tempfile = "3.2.0"
\ No newline at end of file
+tempfile = "3.2.0"
```
Comment (sgb-io) [inline crates/fta/Cargo.toml:27]: think a newline has gone missing - not sure if `cargo fmt` would add it or not.

File: crates/fta/src/lib.rs [hunk 5]
```diff
@@ -1,17 +1,16 @@
 mod config;
 mod cyclo;
 mod halstead;
+pub mod output;
```
[the remark this answers is not in the document]
Comment (bencollins54) [reply crates/fta/src/lib.rs:4]: Removing pub from that line triggers the following:
```
error[E0603]: module `output` is private
 --> crates/fta/src/main.rs:3:10
  |
3 | use fta::output;
  |          ^^^^^^ private module
  ```

File: crates/fta/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bencollins54) [reply crates/fta/src/main.rs:20]: Done.

File: crates/fta/src/lib.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (sgb-io) [reply crates/fta/src/lib.rs:4]: Ok, looked again, it makes sense now. The `output` module is part of the library and needs to be export for usage in `main.rs`. Keeping it `pub` is fine.

**Review by sgb-io — APPROVED** [review]

**Pull request merged**